ui/src/app/views/workspace/component.ts

```diff
@@ -160,18 +164,37 @@ export class WorkspaceComponent implements OnInit, OnDestroy {
     });
   }
 
-  launchCluster(): void {
+  newNotebook(): void {
     if (this.clusterLoading) {
       return;
     }
-    this.localizeNotebooks(this.notebookList).subscribe(() => {
-      // TODO(calbach): Going through this modal is a temporary hack to avoid
-      // triggering pop-up blockers. Likely we'll want to switch notebook
-      // cluster opening to go through a redirect URL to make the localize and
-      // redirect look more atomic to the browser. Once this is in place, rm the
-      // modal and this hacky passing of the launched notebook name.
-      this.launchedNotebookName = '';
-      this.clusterPulled = true;
+    this.localizeNotebooks([]).subscribe(() => {
+      // Use the Jupyter Server API directly to create a new notebook. This
+      // API handles notebook name collisions and matches the behavior of
+      // clicking "new notebook" in the Jupyter UI.
+      // TODO: Use the Swagger generated code instead.
+      const leoNewNotebookUrl =
+        WorkspaceComponent.leoBaseUrl + '/notebooks/' +
+        this.cluster.clusterNamespace + '/' + this.cluster.clusterName +
+        '/api/contents/' + this.clusterLocalDirectory;
+
+      const headers = new Headers();
+      headers.append('Authorization', 'Bearer ' + this.signInService.currentAccessToken);
+      this.http.post(leoNewNotebookUrl, {
```

Comment:
Is this pinging Leo or the notebook cluster? I assume Leo given the todo for swagger generated code?

Reply:
This one goes to the notebook cluster actually. Jupyter has their own swagger API which we can actually call directly on the Jupyter server. Added this link above to make it clearer which API I'm talking about.

https://github.com/jupyter/notebook/blob/master/notebook/services/api/api.yaml